feature/fullstack

johonassnava opened the pull request.

Commit:
Add my readme file

johonassnava committed:
create feat

johonassnava committed:
update api code

johonassnava committed:
add folder test

johonassnava committed:
add retrievePosts logic

johonassnava committed:
add populate file

johonassnava committed:
add mongoose client

johonassnava committed:
add authenticateUser and registerUser logic

johonassnava committed:
add new test folder

johonassnava committed:
Add retrieveUser logic

johonassnava committed:
add logic and test of deletePost, retrievePosts and retrieveUser

johonassnava committed:
add retrieveMyPosts logic

johonassnava committed:
delete message of empty myPosts array

johonassnava committed:
add retrieveSavedPosts and toggleSavePost logic

johonassnava committed:
update logic

johonassnava committed:
add toggleLikePost logic

johonassnava committed:
update userId of retrievePosts.test

johonassnava committed:
update post.likes data type

johonassnava committed:
clean post.like

johonassnava committed:
add retrievePost logic

johonassnava committed:
update retrievePost logic

johonassnava committed:
create updatePost WIP

johonassnava committed:
install express

johonassnava committed:
implement authenticate endpoint

johonassnava committed:
implement createPost api

johonassnava committed:
update createPost route api

johonassnava committed:
add s to create

johonassnava committed:
add retrievePosts toggleLikePost deletePost retrieveUser retrieveMyPosts toggleSavePost retrieveSavedPosts endpoints #43

johonassnava committed:
delete updatePost

johonassnava committed:
add updateUserPassword logic #51

camelCasee commented on staff/johonas-sanchez/fullstack/api/logic/updateUserPassword.js:
```diff
@@ -0,0 +1,54 @@
+        return
+    }
+
+    if (newPassword !== repeatNewPassword) {
```
i think this should be before the database call (User.findById)

johonassnava replied:
Solved

johonassnava committed:
add updateUserPassword logic and endpoint #61

johonassnava committed:
add updateUserEmail logic #56

johonassnava committed:
add updateUserEmail endpoint #62

johonassnava committed:
complet readme endpoints

johonassnava committed:
update Ra

johonassnava committed:
correct validation messages

johonassnava committed:
add fetch conection to registerUser and authenticateUser method

johonassnava committed:
add cors problem solution

johonassnava committed:
add fetch conection files

johonassnava committed:
add cors problem solution

johonassnava committed:
add retrieve posts

johonassnava committed:
add retrievePost endpoint

johonassnava committed:
add retrievePost endpoint

johonassnava committed:
add retrieveUser fetch

johonassnava committed:
add retrieveMyPosts, retrieveSavedPost, updateUserEmail and updateUserPassword fetch, WIP

johonassnava committed:
update fetch gramatical error

johonassnava committed:
add toggleLikePost with fetch and the rest of logic #70

johonassnava committed:
update index html

johonassnava committed:
add app to fullstack folder #71

johonassnava committed:
add functionallity of register user to app #71

johonassnava committed:
add retrieveUser and authenticateUser logic endpoint #71

johonassnava committed:
update registerUser function #71

johonassnava committed:
transform createNewPost to fetch use and add dinamic token #71

johonassnava committed:
integrate retrieve posts and create post in app with api #71

johonassnava committed:
add savedPosts functionality #71

johonassnava committed:
add retrieveMyPosts functionality #71

johonassnava committed:
add deletePost functionality #71

johonassnava committed:
add toggleLikePost #71

johonassnava committed:
add toggleSavePost #71

johonassnava committed:
retrievePosts clean code #71

johonassnava committed:
add correct printing posts order #71

johonassnava committed:
updated names #71

johonassnava committed:
updated names and solve errors #71

johonassnava committed:
solve error with handlePostDeleted #71

johonassnava committed:
add setting page #71

johonassnava committed:
updated setting page #71

johonassnava committed:
add lailwind to project #76

johonassnava committed:
add updated index.css #76

johonassnava committed:
update page style #76

johonassnava committed:
update header style #76

johonassnava committed:
update button style #76

johonassnava committed:
clean code #76

johonassnava committed:
work in userProfile form #76

johonassnava committed:
delete xs block #76

johonassnava committed:
add page reload support in app by means of session storage #79

johonassnava committed:
solve reload error #79

johonassnava committed:
install jsonwebtoken #79

johonassnava committed:
start to use token in every logic #79

johonassnava committed:
solve reload error #79

johonassnava committed:
add updateUserEmail logic a solve toggleLikePost error with postId #79

johonassnava committed:
add jwt validation in all logics and error handling in all components #79

johonassnava committed:
add token expired date #79

johonassnava committed:
add validateJWT to every logic files and error handing #79

johonassnava committed:
add centralized error handling in login and register via app #79

johonassnava committed:
refactor app to accept feedback with button; add getLoggedInUserId for Post compo #79

johonassnava committed:
add functionallity of UserProfile #71

johonassnava committed:
add callback error to updateUserEmail and updateUserPassword #71

johonassnava committed:
add calback null of updateUserEmail and updateUserPassword #71

johonassnava committed:
update tests in api #84

johonassnava committed:
reorganise folder, move previous versions to zzz-api #85

johonassnava committed:
add react-router-dom and integrate it in app #86

johonassnava committed:
add react routes in home for its own components (saved, my-posts, new-post, etc) #86

johonassnava committed:
add react-router-dom and integrate it in UserProfile, UpdateEmail and UpdatePassword #86

johonassnava committed:
group library and components into their own bucket files (index.js) #89

johonassnava committed:
@manuelbarzi
move validators and errors to logic folder; provide context for validators and logic; add logics into their own bucket #89

johonassnava committed:
refactor validators to validate context; refactor logics to logic context; extract cors middleware toutils; all these in api #89

johonassnava committed:
index validate in app helpers; add index in helpers in api #89

johonassnava committed:
move api and app settings to config file #92

johonassnava committed:
move mongodb url to env file #92

johonassnava committed:
delete unuse not used const of users/mail #89

johonassnava committed:
update api package version

johonassnava committed:
add password encryption in api #95

johonassnava committed:
improve error handing #89

johonassnava committed:
add custom errors to improve semantics; add mongo id validation in validate; add missing catch  #89

johonassnava committed:
move route handlers to files #89

johonassnava committed:
apply error handling improvements in app #89

johonassnava committed:
add promise sintax to api registerUser and handler #89

johonassnava committed:
add promise sintax to api authenticateUser and handler #89

johonassnava committed:
add promise sintax to api retrieveUser and handler #89
